src/Database/PostgreSQL/Simple.hs

```diff
@@ -747,7 +747,7 @@ forEachWith_ parser conn template = foldWith_ parser conn template () . const
 -- > instance (ToField a) => ToRow a where
 -- >     ...
 --
--- Unfortunately, this wreaks havoc with type inference, so we take a
+-- Unfortunately, this wreaks havoc on instance resolution, so we take a
```

Comment:
I wouldn't say the change makes this any better. instance resolution is part of type-checking/inference.

Having such instance is a bad idea, if it's not the only instance. And this is not a place to go long on the details.